plugins/inputs/kube_inventory/client.go

```diff
@@ -86,10 +86,14 @@ func (c *client) getIngress(ctx context.Context) (*netv1.IngressList, error) {
 	return c.NetworkingV1().Ingresses(c.namespace).List(ctx, metav1.ListOptions{})
 }
 
-func (c *client) getNodes(ctx context.Context) (*corev1.NodeList, error) {
+func (c *client) getNodes(ctx context.Context, name string) (*corev1.NodeList, error) {
 	ctx, cancel := context.WithTimeout(ctx, c.timeout)
 	defer cancel()
-	return c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	var fieldSelector string
+	if name != "" {
+		fieldSelector = "metadata.name=" + name
+	}
+	return c.CoreV1().Nodes().List(ctx, metav1.ListOptions{FieldSelector: fieldSelector})
```

Comment:
An alternative would be to store the `metav1.ListOptions` object in the client, initialize it once when creating the client and just use it here. This also has the advantage that you don't need to introduce a new parameter in the function... What do you think?

Reply:
we use different parameters when we call nodes(name) and when we call pods(nodeName) so I don't think we can use this idea

Reply:
You are right. Sorry for the noise!

Reply:
Hmmm if I look at the code in both cases you pass in `ki.NodeName` as `name` used in the `fieldselector`.... Am I missing something?

Reply:
yes in one the field selector is 
"spec.nodeName=" + nodeName
and in the other 
"metadata.name=" + name
they are not the same

Reply:
Sorry for missing that! You are absolutely right!